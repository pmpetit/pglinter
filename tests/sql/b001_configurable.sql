-- Quick test to verify B001 rule uses configurable thresholds
BEGIN;

-- Create a table without primary key
CREATE TABLE test_no_pk (
    id INTEGER,
    name TEXT
);

-- Create a table with a primary key
CREATE TABLE test_with_pk (
    id INTEGER PRIMARY KEY,
    name TEXT
);

-- Drop extension if exists to reset state
DROP EXTENSION IF EXISTS pglinter;
-- Create extension
CREATE EXTENSION IF NOT EXISTS pglinter;

-- First, disable all rules to isolate B001 testing
SELECT pglinter.disable_all_rules() AS all_rules_disabled;

-- Enable only B001 for focused testing
SELECT pglinter.enable_rule('B001') AS b001_enabled;

-- Test B001 rule - should show it uses the configured thresholds
SELECT pglinter.perform_base_check();
SELECT pglinter.perform_base_check('/tmp/pglinter_b001_results.sarif');
\! md5sum /tmp/pglinter_b001_results.sarif

-- Update B001 thresholds to very large values (60%, 80%) not to trigger on any table without PK
SELECT pglinter.update_rule_levels('B001', 60, 80);

-- Check updated thresholds
SELECT
    warning_level,
    error_level
FROM pglinter.rules
WHERE code = 'B001';

-- Test B001 rule again - should now trigger with new thresholds
SELECT pglinter.perform_base_check();

-- Update B001 thresholds to very low values (1%, 2%) not to trigger on any table without PK
SELECT pglinter.update_rule_levels('B001', 60, 80);


ROLLBACK;
