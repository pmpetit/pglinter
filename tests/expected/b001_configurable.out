-- Quick test to verify B001 rule uses configurable thresholds
CREATE EXTENSION pglinter;
BEGIN;
-- Create a table without primary key
CREATE TABLE test_no_pk (
    id INTEGER,
    name TEXT
);
-- Create a table with a primary key
CREATE TABLE test_with_pk (
    id INTEGER PRIMARY KEY,
    name TEXT
);
-- First, disable all rules to isolate B001 testing
SELECT pglinter.disable_all_rules() AS all_rules_disabled;
NOTICE:  üî¥ Disabled 20 rule(s)
 all_rules_disabled 
--------------------
                 20
(1 row)

-- Enable only B001 for focused testing
SELECT pglinter.enable_rule('B001') AS b001_enabled;
NOTICE:  ‚úÖ Rule B001 has been enabled
 b001_enabled 
--------------
 t
(1 row)

-- Test B001 rule - should show it uses the configured thresholds
SELECT pglinter.perform_base_check();
NOTICE:  üîç pglinter found 1 issue(s):
NOTICE:  ==================================================
NOTICE:  ‚ö†Ô∏è  [B001] WARNING: 1/2 table(s) without primary key exceed the warning threshold: 50%.
NOTICE:  ==================================================
NOTICE:  üìä Summary: 0 error(s), 1 warning(s), 0 info
NOTICE:  üü° Some warnings found - consider reviewing for optimization
 perform_base_check 
--------------------
 t
(1 row)

SELECT pglinter.perform_base_check('/tmp/pglinter_b001_results.sarif');
 perform_base_check 
--------------------
 t
(1 row)

\! md5sum /tmp/pglinter_b001_results.sarif
47790c72d41ca108840793a8c5977514  /tmp/pglinter_b001_results.sarif
-- Update B001 thresholds to very large values (60%, 80%) not to trigger on any table without PK
SELECT pglinter.update_rule_levels('B001', 60, 80);
NOTICE:  ‚úÖ Updated rule B001 levels: warning=60, error=80
 update_rule_levels 
--------------------
 t
(1 row)

-- Check updated thresholds
SELECT
    warning_level,
    error_level
FROM pglinter.rules
WHERE code = 'B001';
 warning_level | error_level 
---------------+-------------
            60 |          80
(1 row)

-- Test B001 rule again - should now trigger with new thresholds
SELECT pglinter.perform_base_check();
NOTICE:  ‚úÖ No issues found - database schema looks good!
 perform_base_check 
--------------------
 t
(1 row)

-- Update B001 thresholds to very low values (1%, 2%) not to trigger on any table without PK
SELECT pglinter.update_rule_levels('B001', 60, 80);
NOTICE:  ‚úÖ Updated rule B001 levels: warning=60, error=80
 update_rule_levels 
--------------------
 t
(1 row)

ROLLBACK;
DROP EXTENSION pglinter CASCADE;
